### PR TITLE
[PARQUET] Fix a bug converting udt subfield of struct field

### PR DESCRIPTION
**Pull request by changhiskhan**
And also add a `to_pandas` method to rikai.parquet.dataset.Dataset

### CODE REVIEW

**Comment by changhiskhan** [comment]
closes #410

File: python/rikai/parquet/dataset.py [hunk 1]
```diff
@@ -228,3 +225,50 @@ def __iter__(self):
...
+        if not isinstance(field_type, dict):
+            return col
+        elif field_type["type"] == "udt":
+            udt = self._find_udt(field_type["pyClass"])
+            return col.apply(partial(_convert_udt_value, udt=udt))
+        elif field_type["type"] == "struct":
+            return col.apply(lambda d: self._convert(d, field_type))
+        elif field_type["type"] == "array":
+
+            def convert_array(arr):
```
Comment (eddyxu) [inline python/rikai/parquet/dataset.py:256]: shall we move this to be a standalone function globally?

File: python/rikai/parquet/dataset.py [hunk 2]
```diff
@@ -228,3 +225,50 @@ def __iter__(self):
             yield self._convert(
                 shuffler.pop().to_dict(), self.spark_row_metadata
             )
+
+    def to_pandas(self):
```
Comment (eddyxu) [inline python/rikai/parquet/dataset.py:229]: is this user-facing API?  lets add a comment to mention that this method loads all data into memory?

Comment (changhiskhan) [reply python/rikai/parquet/dataset.py:229]: I'll add some docstrings.
I was thinking I can add a `limit` parameter that's just pq.ParquetDataset.head (https://arrow.apache.org/docs/python/generated/pyarrow.dataset.Dataset.html#pyarrow.dataset.Dataset.head) if you only want a preview

File: python/rikai/parquet/dataset.py (hunk 1 above)
Comment (changhiskhan) [reply python/rikai/parquet/dataset.py:256]: yeah, good idea this is usable for the self._convert as well

**Comment by changhiskhan** [comment]
@eddyxu take another look?

**Review by eddyxu — APPROVED** [review]
LGTM. Thanks @changhiskhan

**Pull request merged**